### PR TITLE
Add basic validators & tests

### PR DESCRIPTION
**Pull request by mattmillr**
New PR for this

### CODE REVIEW

File: chassis/util/validators.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    def validate(self, value):
+        match = self.pattern.match(value)
+
+        if match and value == match.group():
+            return value
+
+        self.fail()
+
+
+class Integer(BaseValidator):
```
Comment (rcatlin) [inline chassis/util/validators.py]: Let's have Integer extend Number (ex: `class Integer(Number)`)

File: chassis/test/util/validators_test.py [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+
+    # TODO: Test failure
+    # TODO: Test that validate is not implemented
+
+    # TODO: Test default message
+    # TODO: Test override message
+    # TODO: Test override message in failure
+
+    # TODO: Test default documentation
+    # TODO: Test override documentation
```
Comment (mattmillr) [inline chassis/test/util/validators_test.py]: I'm not using messages yet, so I haven't written these tests. Will write them on the PR for the param parser.

File: chassis/util/validators.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
+"""Parameter Parsing and Validation for Chassis Applications."""
+
+import re
+import six
+
+class ValidationError(Exception):
+    """Raised when validation fails."""
+
+    def __init__(self, message):
```
Comment (mattmillr) [inline chassis/util/validators.py]: This is unnecessary. It's also fixed in the next PR.

**Comment by rcatlin** [comment]
:+1:

**Pull request merged**